pkg/health/health_replicaset.go

```diff
@@ -27,10 +31,19 @@ func getReplicaSetHealth(obj *unstructured.Unstructured) (*HealthStatus, error)
 }
 
 func getAppsv1ReplicaSetHealth(replicaSet *appsv1.ReplicaSet) (*HealthStatus, error) {
+	if time.Since(replicaSet.CreationTimestamp.Time) <= replicaSetBufferPeriod {
```

Comment:
This shouldn't be an early exit - it should be added to ReadinessProbe.InitialDelaySeconds or used as the delay if no readiness probe is defined.  If a replicaset becomes healthy beforehand we shouldn't wait this buffer before updating - its only if its not ready.

Reply:
ah good point.

> it should be added to ReadinessProbe.InitialDelaySeconds

shouldn't we only use the initial delay if readiness hasn't been set?